test/archethic/utils/detect_node_responsiveness_test.exs

```diff
@@ -0,0 +1,290 @@
+defmodule Archethic.Utils.DetectNodeResponsivenessTest do
+  use ArchethicCase
+
+  alias Archethic.Utils.DetectNodeResponsiveness
+
+  import Mox
+  alias Archethic.Crypto
+  alias Archethic.P2P
+  alias Archethic.P2P.Node
+
+  test "start_link/2 for start state" do
+    address = <<0::8, 0::8, :crypto.strong_rand_bytes(32)::binary>>
+
+    replaying_fn = fn count ->
+      count
+    end
+
+    {:ok, pid} = DetectNodeResponsiveness.start_link(address, replaying_fn)
+    assert true == Process.alive?(pid)
+  end
+
+  test "start_link/2 for hard timeout state" do
+    address = <<0::8, 0::8, :crypto.strong_rand_bytes(32)::binary>>
+    # dummy replaying fn with count as argument
+    replaying_fn = fn count ->
+      count
+    end
+
+    {:ok, pid} = DetectNodeResponsiveness.start_link(address, replaying_fn)
```

Comment:
I think you should make the timeouts configurable, as 10s for all tests is too long in a unit testing suite.